src/System.Management.Automation/engine/CoreAdapter.cs

```diff
@@ -4911,7 +4911,8 @@ protected override object PropertyGet(PSProperty property)
         /// <param name="convertIfPossible">instructs the adapter to convert before setting, if the adapter supports conversion</param>
         protected override void PropertySet(PSProperty property, object setValue, bool convertIfPossible)
         {
-            string valueString = setValue as string;
+            // XML is always a string so implicitly convert to string
+            string valueString = LanguagePrimitives.ConvertTo<string>(setValue);
             if (valueString == null)
```

Comment:
I think this condition is unreachable now.

We can convert everything to string (including `$null` which turns into the empty string).

The only failure condition now is if the conversion to string fails with an exception, e.g. if `ToString()` throws an exception. This should be rare, but if it happens, I think we'd want to see that exception and not the `SetValueException` that is being thrown here.

In other words, I think this test and subsequent throw can be safely deleted.